### ISSUE TITLE
Watching with client-go hangs

### ISSUE THREAD

**Issue by ernoaapa** [jetstack/kube-oidc-proxy#146]
Hi,
I have Go CLI that watch CRD resources. For unknown reasons, it's unable to watch the resources.

This line in my go app, never return:
```go
w, err := client.MyCustomResources("default").Watch(metav1.ListOptions{})
```

Should the watch be supported through the `kube-oidc-proxy`?

**Comment by JoshVanL** [comment]
Hi there! 

Looks like my reply never got posted so I'll try again :neutral_face: 

It looks like to me that you are finding the same problem as [here](https://github.com/jetstack/kube-oidc-proxy/issues/135). Watch requests often are not picked up by the httputil.ReverseProxy as long running connections and thus are not flushed immediately. To fix this, the default flush interval (previously never flushed) is now set to 50ms by default which [can be configured](https://github.com/jetstack/kube-oidc-proxy/pull/137). 

This has now been merged and will be contained in v0.3.0 which I want to get released in the coming days.

**Comment by JoshVanL** [comment]
Should be now fixed.

/close
